File: programers_Lv1/시저 암호.js
// https://school.programmers.co.kr/learn/courses/30/lessons/12926?language=javascript
function solution(s, n) {
  const alphaLower = "abcdefghijklmnopqrstuvwxyz";
  const alphaUpper = "ABCDEFGHIJKLMNOPQRSTUVWXYZ";

  const answer = [...s]
    .map((el) => {
      return alphaLower.includes(el)
        ? alphaLower.indexOf(el) + n >= alphaLower.length
          ? alphaLower[alphaLower.indexOf(el) + n - alphaLower.length]
          : alphaLower[alphaLower.indexOf(el) + n]
        : alphaUpper.includes(el)
        ? alphaUpper.indexOf(el) + n >= alphaUpper.length
          ? alphaUpper[alphaUpper.indexOf(el) + n - alphaUpper.length]
          : alphaUpper[alphaUpper.indexOf(el) + n]
        : el;
    })
    .join("");

  return answer;
}

console.log(solution("a B z", 4));
console.log(solution("z", 1));

File: programers_Lv1/시저 암호.py
# https://school.programmers.co.kr/learn/courses/30/lessons/12926?language=python3
from string import ascii_lowercase, ascii_uppercase

def solution(s, n):
    answer = ''
    for c in s:
        if c in ascii_lowercase:
            idx = ascii_lowercase.find(c)+n if len(ascii_lowercase) > ascii_lowercase.find(c)+n else ascii_lowercase.find(c)+n - len(ascii_lowercase)
            answer += ascii_lowercase[idx]
        elif c in ascii_uppercase:
            idx = ascii_uppercase.find(c)+n if len(ascii_uppercase) > ascii_uppercase.find(c)+n else ascii_uppercase.find(c)+n - len(ascii_uppercase)
            answer += ascii_uppercase[idx]
        else:
            answer += c

    return answer

print(solution("a B z", 4))
print(solution("z", 1))
